{{cookiecutter.project_slug}}/_/ci-services/deploy-stage/bitbucket-pipelines.yml

```diff
@@ -32,35 +33,29 @@
       image: docker.io/appuio/oc:v3.11
       script:
       - TARGET={{ cookiecutter.cloud_project }}{% if cookiecutter.environment_strategy == 'dedicated' %}-${BITBUCKET_DEPLOYMENT_ENVIRONMENT}{% endif %}
-        REVIEW_APP=review-pr${BITBUCKET_PR_ID}
-        APPLICATION=${REVIEW_APP}-application
-        DATABASE_HOST=${REVIEW_APP}-{{ cookiecutter.database|lower }}
+        LABEL=review-pr${BITBUCKET_PR_ID}
+        APPLICATION=application-${LABEL}
+        DATABASE_HOST={{ cookiecutter.database|lower }}-${LABEL}
       - oc login ${KUBE_URL} --token="${KUBE_TOKEN}" -n ${TARGET}
       - *generate-secrets-vars
       - *generate-secrets-app
       - *generate-secrets-db
       - pushd deployment/application/base &&
         kustomize edit set image IMAGE="docker-registry.default.svc:5000/${TARGET}/{{ cookiecutter.project_slug }}:${BITBUCKET_COMMIT}" &&
         popd
-      - pushd deployment/application/overlays/${BITBUCKET_DEPLOYMENT_ENVIRONMENT} &&
-        kustomize edit add label app:${REVIEW_APP} -f &&
-        kustomize edit set nameprefix "${REVIEW_APP}-" &&
-        popd
-      - pushd deployment/database/overlays/${BITBUCKET_DEPLOYMENT_ENVIRONMENT} &&
-        kustomize edit add label app:${REVIEW_APP} -f &&
-        kustomize edit set nameprefix "${REVIEW_APP}-" &&
-        popd
+      - sed "s|REVIEW-ID|${LABEL}|" -i deployment/application/overlays/${BITBUCKET_DEPLOYMENT_ENVIRONMENT}/kustomization.yaml
```

Comment:
Why are you using `sed` here instead of `kustomize edit`?

Reply:
I tried to find a way with `kustomize edit`, but it doesn't seem allow to update the `nameSuffix` and `commonLabels`, or not the way we would need it. I'm sure I tried hard, but failed to find a way. I'm happy to use `kustomize edit` if that's (cleanly) possible.

- [application labels](https://github.com/painless-software/painless-continuous-delivery/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/_/deployment/application/overlays/development/kustomization.yaml#L4-L11)
- [database labels](https://github.com/painless-software/painless-continuous-delivery/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/_/deployment/database/overlays/development/kustomization.yaml#L4-L11)
- `sed` [with GitLab CI](https://github.com/painless-software/painless-continuous-delivery/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/_/ci-services/definitions/.gitlab-ci.yml#L96-L97)

Reply:
What's wrong with `kustomize edit set namesuffix`?
It seems to have worked before, right?

Reply:
`kustomize edit set namesuffix` would work, but there is no `kustomize edit set commonlabels`. And we need to set the `app` label dynamically, too.

Reply:
Cool! `kustomize edit add label` does what we need to do! – That actually makes sense:

- `set` for a single item (`nameSuffix`)
- `add` for a list (`commonLabels`) :+1:

I'll change that immediately. I like the code much better now. Thank you, @srueg!